mitodl/open-discussions
# Wider widget sidebar; more consistent breakpoints

#### Pre-Flight checklist


#### What are the relevant tickets?
#3737

#### What's this PR do?
This PR:
- Fixes the vertical centering of app header
- Makes the widget column in field pages a bit wider, and displays 3, 2, or 1 courses in the carousel depending on the current  [MUI Breakpoint](https://mui.com/material-ui/customization/breakpoints/)
- Additionally, I tried to make our breakpoint usage a bit more consistent within Infinite Corridor. (*There are some behavior changes here, but they seem OK to me. (E.g., the search page used to "go mobile" at 998px; now it "goes mobile" at 839px.*)

#### Any background context you want to provide?
"Breakpoints" tell the website to respond different at screen widgets. Usually, breakpoints are implemented with CSS [media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries). Our use of breakpoints has been a bit inconsistent:
- In Open Discussions, we use three breakpoint systems:
    1. material web, via classes like `mdc-layout-grid__cell--span-8`
    2. our own breakpoints, via sass mixin, e.g.,  `breakpoint(desktop)`
    3. a JS help `useDeviceCategory` which has somewhat different breakpoint values than our CSS
- In Infinite Corridor, we haven't been entirely consistent, but IMO we should try and stick with MUI's breakpoint system (the values of which we can customize if we like).
     1. We are already using MUI's Grid component instead of stuff like `mdc-layout-grid__cell--span-8` anyway
     2. We may still want a sass mixin, which I've added here.
     3. We can use MUI's `useMediaQuery` hook instead of `useDeviceWidth`.

#### How should this be manually tested?
Load a field page (with widgets) and the search page and check that stuff looks reasonable at different widths.


#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/9010790/190007675-f360afed-f843-4ec8-9996-1b0a52fb33f6.mov

<img width="1725" alt="Screen Shot 2022-09-13 at 4 57 44 PM" src="https://user-images.githubusercontent.com/9010790/190007931-d93ee593-9cc3-44bc-9560-d4fa2d8e01b5.png">

**Search Page:** Minor changes, really just using MUI's breakpoints

<img width="1800" alt="Screen Shot 2022-09-13 at 4 32 06 PM" src="https://user-images.githubusercontent.com/9010790/190008023-7003e8c4-5295-4b16-aefb-7f4e7dcf7a0a.png">

<img width="300" alt="Screen Shot 2022-09-15 at 4 45 07 PM" src="https://user-images.githubusercontent.com/9010790/190507708-2b62731d-ae61-40f1-a892-a907b8547414.png"> <img width="300" alt="Screen Shot 2022-09-15 at 5 03 25 PM" src="https://user-images.githubusercontent.com/9010790/190508928-0afd22cc-6b2e-4cd4-9169-1534e82f7640.png">
